### ISSUE TITLE
Livecheck PageMatch strategy appears to be broken

### ISSUE THREAD

**Issue by thecesrom** [Homebrew/brew#11882]
### `brew config` output

```shell
HOMEBREW_VERSION: 3.2.8-19-g8d2811e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8d2811ea71e0d121cb0b06e160ddb1680c081d68
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5314b376a917de9bd4ed145e94f54ec0013706a7
Core tap last commit: 25 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BAT: set
HOMEBREW_CASK_OPTS: []
HOMEBREW_CLEANUP_MAX_AGE_DAYS: 3
HOMEBREW_EDITOR: nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_GIT_EMAIL: thecesrom@gmail.com
HOMEBREW_GIT_NAME: César Román
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: quad-core 64-bit ivybridge
Clang: 12.0.0 build 1200
Git: 2.33.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.0.32.29
Xcode: N/A
```

### `brew doctor` output

```shell
﻿Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/thecesrom/.pyenv/shims/python2-config
  /Users/thecesrom/.pyenv/shims/python2.7-config
  /Users/thecesrom/.pyenv/shims/python3.9-config
  /Users/thecesrom/.pyenv/shims/python-config
  /Users/thecesrom/.pyenv/shims/python3-config
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I tried to run `brew livecheck` and some casks are failing. All of them are using the `PageMatch` strategy extracting certain fields.

My `.brew_livecheck_watchlist` file looks like this:

```
azure-data-studio
azure-data-studio-insiders
black
flake8
isort
pre-commit
pycharm-ce
pyenv
rectangle
tunnelbear
visual-studio-code
visual-studio-code-insiders
```

### What happened (include all command output)?

```bash
$ brew livecheck
azure-data-studio : 1.31.1 ==> 1.32.0
Error: azure-data-studio-insiders: Unable to get versions
black : 21.7b0 ==> 21.7b0
flake8 : 3.9.2 ==> 3.9.2
isort : 5.9.3 ==> 5.9.3
pre-commit : 2.14.0 ==> 2.14.0
Error: pycharm-ce: Unable to get versions
pyenv : 2.0.4 ==> 2.0.4
rectangle : 0.48,53 ==> 0.48,53
tunnelbear : 4.1.6,1614795843 ==> 4.1.7,1628781930
visual-studio-code : 1.59.0 ==> 1.59.0
Error: visual-studio-code-insiders: Unable to get versions
```

I also found other casks using the same strategy:

```bash
$ brew livecheck --debug alfred clion steam unity azure-data-studio-insiders visual-studio-code-insiders

Cask:             alfred
Livecheckable?:   Yes

URL:              https://www.alfredapp.com/app/update4/general.xml
Strategy:         PageMatch
Error: alfred: Unable to get versions

----------

Cask:             azure-data-studio-insiders
Livecheckable?:   Yes

URL:              https://azuredatastudio-update.azurewebsites.net/api/update/darwin/insider/VERSION
Strategy:         PageMatch
Error: azure-data-studio-insiders: Unable to get versions

----------

Cask:             clion
Livecheckable?:   Yes

URL:              https://data.services.jetbrains.com/products/releases?code=CL&latest=true&type=release
Strategy:         PageMatch
Error: clion: Unable to get versions

----------

Cask:             steam
Livecheckable?:   Yes

URL:              https://store.steampowered.com/oldnews/?feed=steam_client
Strategy:         PageMatch
Error: steam: Unable to get versions

----------

Cask:             unity
Livecheckable?:   Yes

URL:              https://public-cdn.cloud.unity3d.com/hub/prod/releases-darwin.json
Strategy:         PageMatch
Error: unity: Unable to get versions

----------

Cask:             visual-studio-code-insiders
Livecheckable?:   Yes

URL:              https://update.code.visualstudio.com/api/update/darwin-universal/insider/VERSION
Strategy:         PageMatch
Error: visual-studio-code-insiders: Unable to get versions
```

### What did you expect to happen?

I expected the command to succeed and check if the versions match or must be updated.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. Run `brew livecheck alfred azure-data-studio-insiders`
2. Have command succeed checking version
```

**Comment by MikeMcQuaid** [comment]
CC @samford in case this relates to recent work

**Comment by samford** [comment]
I looked into this and it was introduced in 26821301e74b3b4a559877c97ce8af1b8da8f545 and relates to the added early return conditions in `PageMatch#find_versions`. I had tested this on homebrew/core without any issue but I didn't test it on homebrew/cask.

The basic issue here is that `PageMatch` requires a regex but a number of `livecheck` blocks in casks don't adhere to this because the regex is inlined into the `strategy` block. I've warned against this in general and some newer homebrew/cask `livecheck` blocks use `#regex` and pass the value into the `strategy` block but there are still a number of older `PageMatch` `strategy` blocks where the regex is inlined (e.g., `alfred`). These are the ones that are currently broken.

I'll open a PR to update the early return conditions to require `url` and either `regex` or a block, so the checks that are currently broken will work again.

In the long run, a lot of the `livecheck` blocks in casks with simple `strategy` block like `alfred` should move the regex into a `#regex` call and pass it into the strategy block. This is the standard pattern in homebrew/core and it's starting to get picked up in homebrew/cask but there are still a lot of older `strategy` blocks that need to be updated.

That said, there are some cask `strategy` blocks where more than one regex is required to identify the full version and neither of the regexes can be viewed as the primary regex. Those are exceptional cases where it's fine to inline the regexes in the `strategy` block and the updated early return conditions will support that scenario.
